spec/requests/admin_spec.rb

```diff
@@ -17,23 +17,43 @@
           "provider" => "dfe",
           "info" => {"email" => "test-dfe-sign-in@host.tld"}
         )
-        post admin_dfe_sign_in_path
-        follow_redirect!
       end
 
-      it "renders the admin page and sets a session" do
-        get admin_path
+      context "when the user is authorised to access the service" do
+        before do
+          allow_any_instance_of(DfeSignin::Authorisation).to receive(:has_role?) { true }
```

Comment:
As in the feature-spec, we shouldn't be stubbing our own internal code. It should be possible to set the auth response using `OmniAuth.config.mock_auth[:dfe]=` so that it returns a response with the appropriate role in it.